pkg/v1/tkg/managementcomponents/helper.go

```diff
@@ -21,18 +23,18 @@ const (
 )
 
 // GetTKGPackageConfigValuesFileFromUserConfig returns values file from user configuration
-func GetTKGPackageConfigValuesFileFromUserConfig(managementPackageVersion string, userProviderConfigValues map[string]interface{}) (string, error) {
+func GetTKGPackageConfigValuesFileFromUserConfig(managementPackageVersion string, userProviderConfigValues map[string]interface{}, tkgBomConfig *tkgconfigbom.BOMConfiguration) (string, error) {
 	// TODO: Temporary hack(hard coded values) to configure TKR source controller package values. This should be replaced with the logic
 	// that fetches these values from tkg-bom(for bom related urls) and set the TKR source controller package values
 	var tkrRepoImagePath string
 	providerType := userProviderConfigValues[constants.ConfigVariableProviderType]
 	switch providerType {
 	case constants.InfrastructureProviderVSphere:
-		tkrRepoImagePath = "projects-stg.registry.vmware.com/tkg/tkr-repository-vsphere-nonparavirt"
+		tkrRepoImagePath = fmt.Sprintf("%s/%s", tkgBomConfig.ImageConfig.ImageRepository, tkgBomConfig.TKRPackageRepo.VSphereNonparavirt)
```

Comment:
I think this is an additional fix we did to leverage the bom data for source controller parameters. Maybe add a note of that in the PR description, commit message.